src/movies.js

```diff
@@ -2,14 +2,122 @@
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
 
+function orderByYear(newArr) {
+  let array2 = [...newArr];
+  let orderByYear = array2.sort(function(objA, objB) {
+    if (objA.year < objB.year) {
+      return -1;
+    }
+    if (objA.year > objB.year) {
+      return 1;
+    }
+    if (objA.year === objB.year) {
+      if (objA.title < objB.title) {
+        return -1;
+      }
+      if (objA.title > objB.title) {
+        return 1;
+      }
+      if (objA.title === objB.title) {
+        return 0;
+      }
+    }
+  });
+  return orderByYear;
+}
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+/*Go ahead and create a howManyMovies() function that receives an array as a parameter and filter 👀 
+the array so we can have only the drama movies where Steven Spielberg is the director.
+*/
+
+function howManyMovies(array) {
+  /* if (num=0){
+
+    };
+    return num
+    */
+  if (array.length === 0) {
+    return 0;
+  }
+  const newArray = array.filter(function(obj, i, originalArr) {
+    if (obj.director === "Steven Spielberg" && obj.genre.includes("Drama")) {
+      return true;
+    }
+  });
+
+  return newArray.length;
+}
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically(newArray) {
+  let array3 = [...newArray];
```

Comment:
Avoid using names for your variables like array3 and be more specific with naming. In this case you could use array for the function parameter and arrayCopy for the variable, for example.